lists/downloader.go

```diff
@@ -145,6 +146,16 @@ func (d *HTTPDownloader) DownloadFile(link string) (io.ReadCloser, error) {
 			onDownloadError(link)
 		}))
 
+	var transientErr *TransientError
```

Comment:
Do we need it here? Downloader should just return the error without wrapping/unwrapping. Otherwise the code is polluted with not necessary things. This is a task for caller (in our case the test).

Reply:
Depends on the premise.
I implemented it with the test first premise.
If the premise is implementation first the code can be moved to the test go file.
Main cause of flakiness in this test is that an dnsError is expected but a transient error of type dnsError could be returned if executed in race.

Reply:
Reworked it by unwrapping transient error in Test instead of downloader.